spring-cloud-aws-messaging/src/test/java/io/awspring/cloud/messaging/endpoint/NotificationMessageHandlerMethodArgumentResolverTest.java

```diff
@@ -86,4 +95,25 @@ void supportsParameter_withIntegerParameterType_shouldReturnFalse() throws Excep
 		assertThat(supportsParameter).isTrue();
 	}
 
+	@Test
+	void resolveArgument_notificationMessageTypeWithSubject_reportsErrors_failsVerification() throws Exception {
+		// Arrange
+		SnsMessageManager snsMessageManager = new SnsMessageManager("eu-east-1");
+		NotificationMessageHandlerMethodArgumentResolver resolver = new NotificationMessageHandlerMethodArgumentResolver(
+				snsMessageManager);
+
+		byte[] subscriptionRequestJsonContent = FileCopyUtils
+				.copyToByteArray(new ClassPathResource("notificationMessage.json", getClass()).getInputStream());
+		MockHttpServletRequest servletRequest = new MockHttpServletRequest();
+		servletRequest.setContent(subscriptionRequestJsonContent);
+
+		MethodParameter methodParameter = new MethodParameter(
+				ReflectionUtils.findMethod(NotificationMethods.class, "handleMethod", String.class, String.class), 0);
+
+		// Assert
+		assertThatThrownBy(() -> resolver.resolveArgument(methodParameter, null, new ServletWebRequest(servletRequest),
+				null)).isInstanceOf(SdkClientException.class).hasMessageContaining(
+						"igningCertUrl does not match expected endpoint. Expected sns.eu-east-1.amazonaws.com but received endpoint was sns.eu-west-1.amazonaws.com.");
```

Comment:
```suggestion
						"signingCertUrl does not match expected endpoint. Expected sns.eu-east-1.amazonaws.com but received endpoint was sns.eu-west-1.amazonaws.com.");
```

Reply:
Or uppercase "S"?

Reply:
> Can you add a sample app for SNS so that we can easily spin it up and verify?

https://github.com/awspring/spring-cloud-aws/blob/2.4.x/spring-cloud-aws-samples/spring-cloud-aws-sns-sample/src/main/java/io/awspring/cloud/sns/sample/NotificationMappingController.java
Sample already has http controller